src/v0/destinations/firehose/transform.js

```diff
@@ -6,7 +6,7 @@ function getDeliveryStreamMapTo(event) {
   const { mapEvents } = destination.Config;
   const hashMap = getHashFromArray(mapEvents, 'from', 'to');
   return (
-    (message.event ? hashMap[message.event.toLowerCase()] : null) ||
+    (message.event ? hashMap[message?.event.toLowerCase()] : null) ||
```

Comment:
We are already checking validity of event. Why do we need optional chaining here? 
Maybe this issue could be type issue maybe `event` is not a string

Reply:
In metadata I see that eventName is an empty string but with that this check would not have failed. Should I remove it? The error was intermittent for a short while and not occurring now.

Reply:
Better to check  `(isString(message.event) and message.event.length != 0 ? hashMap[message?.event.toLowerCase()] : null)`